WordPress/gutenberg
# [Mobile] - Fix regression with the Color hook and ColorPanel

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5683

## What?
This PR is a follow-up of https://github.com/WordPress/gutenberg/pull/48893 that refactored the usage of the `Color` hook and the `ColorPanel` component. These changes broke the Text and Background color customizations in the mobile editor.

## Why?
To bring back the Color and Background color customization functionality to the mobile editor.

## How?

It moves the native `ColorPanel` to the `components/global-styles` to keep the same structure as the Web editor, it brings the web changes and keeps the current functionality we had for the ContrastChecker.

Removes current code like [avoiding manually resetting](https://github.com/WordPress/gutenberg/pull/49917/commits/7c6a8e4f4c0ff3b552f2e6e5a27a629ef7833c0a) color changes since it will be managed by the ColorPanel. 

Adds new `getGlobalStyles` [integration helper](https://github.com/WordPress/gutenberg/pull/49917/commits/fec37660132e6f07d68d180b5137f1e6c318f5c3) that can be used with `initializeEditor` passing `withGlobalStyles` as `true` which will pass to the editor some Global styles data. This can be expanded in the future depending on the requirements.

It also adds the following tests to prevent future breakages related to colors customizations:

**Paragraph block:**
- Sets a text color
- Sets a background color
- Sets a text and background color
- Clears selected text and background color
- Checks gradient option is not shown
- Sets a theme text color
- Checks the ContrastCheck is shown

**Heading block:**
- Sets a text color
- Sets a background color

**Buttons block:**
- Sets a text color
- Sets a background color
- Sets a gradient background color

## Testing Instructions

- Open the editor
- Add different blocks and check the following:
  - **Paragraph block**: Text and Background (No gradients options) 
  - **Heading block**: Text and Background (No gradients options)
  - **Quote block**: Text and Background (No gradients options)
  - **List block**: Text and Background (No gradients options)
  - **Columns block**: Text and Background (No gradients options)
  - **Buttons block**: Text and Background (**With** gradients options)
  - **Cover block**: No Background options, only Overlay customization, and Text.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

<video src="https://user-images.githubusercontent.com/4885740/233141935-32c02f03-05b5-4097-878b-3c493b7e9eab.mov" width=200/>
